### PR TITLE
Add strict types to tracking context

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Refactor tracking context to specify the types explicitly

## Justify why they are needed

- This makes it easier to use the values from the context without having to cast them to the correct type. It's also more type safe.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3641** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3641?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3642** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3642?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3643** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3643?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #3644** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3644?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #3645** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3645?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #3646** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3646?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #3647** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3647?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #3648** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3648?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #3651** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3651?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #3652** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3652?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **PR #3653** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3653?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                        * **PR #3654** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3654?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                          * **PR #3655** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3655?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                            * **PR #3656** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3656?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                              * **PR #3657** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3657?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                * **PR #3658** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3658?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3658?utm_source=stack-comment).

**Review by alebedev — APPROVED** [review]
Nice!  Looks easier to use this way

File: apps/store/src/services/Tracking/TrackingContext.tsx [hunk 1]
```diff
@@ -37,18 +37,18 @@ export const TrackingProvider = (props: Props) => {
...
 })
 
 const parsePriceIntentData = (data: Record<string, unknown>): Tracking['context'] => {
   const numberCoInsured = parseNumber(data.numberCoInsured)
 
   return {
     numberOfPeople: numberCoInsured ? numberCoInsured + 1 : undefined,
-    zipCode: data.zipCode,
-    city: data.city,
+    zipCode: typeof data.zipCode === 'string' ? data.zipCode : undefined,
```
Comment (gustaveen) [inline apps/store/src/services/Tracking/TrackingContext.tsx:50]: Would `data.zipCode ?? undefined` not work in this case?

**Review by gustaveen — APPROVED** [review]
Perfect!

File: apps/store/src/services/Tracking/TrackingContext.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/services/Tracking/TrackingContext.tsx:50]: Well typescript will complain since the value is "unknown" to begin with

**Comment by robinandeer** [comment]
### Merge activity

* **Nov 29, 11:01 AM**: @robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3658).

**Pull request merged**